static/js/fetch_status.js

```diff
@@ -30,7 +31,14 @@ const FetchStatus = function () {
         loading_newer = true;
     };
 
-    self.finish_newer_batch = function (opts) {
+    self.finish_newer_batch = function (messages, opts) {
+        let found_max_message_id = 0;
+        for (const msg of messages) {
```

Comment:
We seem to duplicate this idiom twice.   Does it make sense to have little helper called `max_id_for_messages`?